### PR TITLE
[CLA Approved] Gauge bug fix #6593: Value range and limit indicators and button disa…

### LINKED ISSUE #6593: [Gauge] Value range and limit indicators are broken
#### Summary

When a user manually enters upper/lower limits, the validity indicators (the red x or green check) that used to work, do not anymore. They stay showing invalid, regardless of whether the values a user inputs are valid or not. Additionally, when a user *does* enter an invalid limit, the OK button used to be disabled, disallowing users to save the Gauge until a valid value was inputted. This doesn't happen anymore– instead, users can save the object with invalid limits. 

Notice in the picture below: The upper limit of 90 *IS* valid if the maximum value is 100. However, the indicator shows it as invalid. 
Additionally, the lower limit of -20 *IS NOT* valid if the minimum value is 0. The indicator shows it as invalid as well. Now notice that the OK button is not disabled, and allows for a user to save the changes. 
Screen Shot 2023-04-13 at 1 10 52 PM
#### Steps to Reproduce

1. Navigate to/ create a gauge.
2. Toggle off the ability to use Telemetry limits.
3. Manually enter in your own limits. Test with invalid and valid imits. For limits to be valid, the lower limit must be higher than the minimum value and the upper limit must be lower than the maximum. For a limit to be invalid, the lower limit should be less than the minimum, and upper limit should be more.
4. Observe that regardless of the limits being valid or not, the validity indicator stays _invalid_. 
5. Observe that when you enter an invalid limit, you are still able to press OK and save the object. 

#### Environment

* Open MCT Version:  2.2.1
* Deployment Type:  Testathon
* OS:
* Browser: Chrome

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [x ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by sinisam92**
This change resolves bug fix #6593 by adding additional validations for value ranges and limit indicators and disables "OK" button depending on validation.  
        - Added additional validation for min value and limitHigh in GaugePlugin.js
        - Added additional validation inside validateRow function in FormRow.vue
        - Stored additional values inside data() on GaugeFormController.vue to be able to pass that data to GaugePlugin.js to be able to validate.

PS: sorry if I have wrote this wrong it is my first :) 

Closes 

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [ ] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by ozyx** [comment]
Hi @sinisam92!
Thanks so much for opening this PR! If you haven't already, please complete and submit a [Contributor License Agreement (CLA)](https://github.com/nasa/openmct/discussions/3821). This is required before we can approve/merge your code.  Thanks again, and we will be reviewing this very soon!

**Comment by sinisam92** [comment]
Hi @ozyx!
I have completed and submited CLA.

**Comment by ozyx** [comment]
> Hi @ozyx! I have completed and submited CLA.

Awesome!

File: src/plugins/gauge/GaugePlugin.js [hunk 1]
```diff
@@ -145,11 +145,11 @@ export default function () {
                             limitHigh: true
                         };
 
-                        if (min === '') {
+                        if (min === '' || min < 0) {
```
Comment (sinisam92) [inline src/plugins/gauge/GaugePlugin.js:148]: I have just sow image of the gauges and they have negative minimum value, so to fix this just delete `min < 0` to fix that problem.

File: src/api/forms/components/FormRow.vue [hunk 2]
```diff
@@ -131,7 +131,12 @@ export default {
                     && data.value !== '';
             }
 
-            if (this.row.required && !valid) {
+            if (this.row.required && !valid
+                    || data.min < 0
```
Comment (sinisam92) [inline src/api/forms/components/FormRow.vue:135]: I have just sow image of the gauges and they have negative minimum value, so to fix this just delete `data.min < 0` to fix that problem of disabled "OK" button.

**Comment by ozyx** [comment]
Closing as we will not be taking this change at this time. Thank you!

**Pull request closed without merging**